pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -246,6 +246,11 @@ type Controller struct {
 	workloadInstancesByIP map[string]*model.WorkloadInstance
 	// Stores a map of workload instance name/namespace to address
 	workloadInstancesIPsByName map[string]string
+	// serviceToWorkloadNodesMap is a mapping of NodePort services to the list of kubernetes node names
+	// on which there is at least one workload belonging to the service is running. This is needed to handle
+	// node-port services with external traffic policy type set to Local (default is Cluster). This is because
+	// when it is set to local and traffic is sent to a node not hosting a workload, it will be dropped
+	serviceToWorkloadNodesMap map[host.Name]map[string]struct{}
```

Comment:
Does it contain services  with Local externalTrafficPolicy only? Atleast That is what I understood based on my code reading. If yes, can you update comments?

Reply:
Can we also make the name reflect this is used specifically for network gateways (service is labeled `topolgy.istio.io/network=something`) and not all NodePort services in the cluster

Reply:
At first glance i could not know what does this mean, will read through

Reply:
@stevenctl - I changed it to `nodePortGwLocalServiceToWorkloadNodesMap`. It is long and wordy. Please feel free to suggest better alternatives 

@ramaraochavali - Yes. It is for those NodePort gateway services with external traffic policy set to `Local`. And yes, I updated comments.
```go
	// nodePortGwLocalServiceToWorkloadNodesMap is a mapping of NodePort services with external traffic policy set to Local
	// to the list of kubernetes node names on which there is at least one workload belonging to the service is
	// running. This is needed to handle node-port services with external traffic policy type set to Local
	// (default is Cluster). This is because when it is set to local and traffic is sent to a node not hosting
	// a workload, it will be dropped
	nodePortGwLocalServiceToWorkloadNodesMap map[host.Name]map[string]struct{}
```